pkg/app/piped/imagewatcher/watcher.go

```diff
@@ -156,51 +155,63 @@ func (w *watcher) determineUpdates(ctx context.Context, repoID string, repo git.
 	if !ok {
 		return nil, fmt.Errorf("configuration file for Image Watcher not found: %w", err)
 	}
-
-	updates := make([]config.ImageWatcherTarget, 0)
-	for _, target := range cfg.Targets {
-		if provider.Name() != target.Provider {
-			continue
-		}
-		outdated, err := checkOutdated(ctx, target, repo, provider)
-		if err != nil {
-			return nil, fmt.Errorf("failed to check the image is outdated: %w", err)
-		}
-		if outdated {
-			updates = append(updates, target)
-		}
-	}
-	return updates, nil
+	return cfg, nil
```

Comment:
nit: I suggest we could simplify the function signature of this `config.LoadImageWatcher` function to 
```go
func LoadImageWatcher(repoRoot string, includes, excludes []string) (*ImageWatcherSpec, error)
```
and create a new error type `config.NotFound`, then check it here with `errors.Is`.
The reasons is
1. current `ok` value here look quite weird, should it be `found`? such that it's clarifying show what do we use it for.
2. I read the implementation of `config.LoadImageWatcher` once, looks like after check for `os.IsNotExist` we have to pass `false` value for it on all other cases.

What do you think of it? 😄

Reply:
Thank you! No objection to that! Other part should use that error type, so let me address it as another patch: https://github.com/pipe-cd/pipe/issues/1259

Reply:
thanks 👍